worker/sshserver/server_test.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package sshserver_test
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"net"
+	"time"
+
+	"github.com/juju/testing"
+	"golang.org/x/crypto/ssh"
+	"google.golang.org/grpc/test/bufconn"
+	gc "gopkg.in/check.v1"
+
+	"github.com/juju/juju/worker/sshserver"
+)
+
+type sshServerSuite struct {
+	testing.IsolationSuite
+
+	userSigner  ssh.Signer
+	jumpHostKey string
+}
+
+var _ = gc.Suite(&sshServerSuite{})
+
+func (s *sshServerSuite) SetUpSuite(c *gc.C) {
+	// Setup user signer
+	userKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	c.Assert(err, gc.IsNil)
+
+	userSigner, err := ssh.NewSignerFromKey(userKey)
+	c.Assert(err, gc.IsNil)
+
+	s.userSigner = userSigner
+
+	// Setup jump host private key
+	jumpHostKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	c.Assert(err, gc.IsNil)
+
+	jumpHostKeyPEM := pem.EncodeToMemory(&pem.Block{
+		Type:  "RSA PRIVATE KEY",
+		Bytes: x509.MarshalPKCS1PrivateKey(jumpHostKey),
+	})
+
+	s.jumpHostKey = string(jumpHostKeyPEM)
+
+}
+
+func (s *sshServerSuite) TestSSHServer(c *gc.C) {
```

Comment:
A lot of the code in this test will be reused across tests. I suggest making an sshtest package that holds the common bits and keeps the tests more focused.